### PR TITLE
[MOBILEAPPS-1622] Redirection code added to check if app already exists in phone i.e an…

### PR DESCRIPTION
**Pull request by jatin2008**
…droid or iphone otherwise opens in the browser

**Please check if the PR fulfills these requirements**

> - [ x] The commit message follows our [guidelines](https://github.com/Alfresco/alfresco-ng2-components/wiki/Commit-format)
> - [ ] Tests for the changes have been added (for bug fixes / features)
> - [ ] Docs have been added / updated (for bug fixes / features)

**What kind of change does this PR introduce?** (check one with "x")

> - [ ] Bugfix
> - [ x] Feature
> - [ ] Code style update (formatting, local variables)
> - [ ] Refactoring (no functional changes, no api changes)
> - [ ] Build related changes
> - [ ] Documentation
> - [ ] Other... Please describe:

**What is the current behaviour?** (You can also link to an open issue here)
Same behaviour as it was running earlier.

**What is the new behaviour?**
I have added code that will check whether device is Android or Iphone and if it encounters the android or iphone device then it will open the link in that application in the phone otherwise it will open the link in the browser.

**Does this PR introduce a breaking change?** (check one with "x")

> - [ ] Yes
> - [ x] No

If this PR contains a breaking change, please describe the impact and migration path for existing applications: ...

**Other information**:

### CODE REVIEW

File: app/src/app/content-plugin/components/shared-link-view/shared-link-view.component.ts [hunk 1]
```diff
@@ -46,6 +46,8 @@ export class SharedLinkViewComponent implements OnInit, OnDestroy {
   private sharedLinksApi: SharedlinksApi;
   sharedLinkId: string = null;
   viewerToolbarActions: Array<ContentActionRef> = [];
+  sharedId: string = null;
```
Comment (DenysVuika) [inline app/src/app/content-plugin/components/shared-link-view/shared-link-view.component.ts:49]: why only shared link view has this functionality?

File: app/src/app/content-plugin/components/shared-link-view/shared-link-view.component.ts [hunk 2]
```diff
@@ -57,6 +59,46 @@ export class SharedLinkViewComponent implements OnInit, OnDestroy {
...
+    this.route.params.subscribe((data) => {
+      this.sharedId = data.id;
+    });
+
+    const ua = navigator.userAgent.toLowerCase();
+    const isAndroid = ua.indexOf('android') > -1;
+    const isIphone = ua.indexOf('iphone') > -1;
+
+    if (isIphone === true) {
+      window.location.href = 'com.alfresco.contentapp://iosamw';
```
Comment (DenysVuika) [inline app/src/app/content-plugin/components/shared-link-view/shared-link-view.component.ts:78]: this should be coming from configuration

File: app/src/app/content-plugin/components/shared-link-view/shared-link-view.component.ts [hunk 3]
```diff
@@ -57,6 +59,46 @@ export class SharedLinkViewComponent implements OnInit, OnDestroy {
...
+      setTimeout(() => {
+        this.previewData();
+      }, 25);
+    } else if (isAndroid === true) {
+      window.location.href = 'com.alfresco.content.app://androidamw://' + this.baseUrl + this.sharedId;
```
Comment (DenysVuika) [inline app/src/app/content-plugin/components/shared-link-view/shared-link-view.component.ts:83]: this should be coming from configuration

**Review by DenysVuika — CHANGES_REQUESTED** [review]

**Comment by DenysVuika** [comment]
@jatin2008 please use the proper format of the PR title with the reference to the corresponding Jira. Also, this PR does not make much sense at you change the "shared link" viewer component only, and hardcode the stuff that should not be there

**Comment by jatin2008** [comment]
revoking

**Pull request closed without merging**